### PR TITLE
Hail conda recipe and CI to build it

### PR DESCRIPTION
**Pull request by vladsaveliev**
Adding a conda package recipe for the `cpg` channel, and a GitHub Actions workflow to build it and push to anaconda. It should trigger on pushes to main. Long term, we could reconsider and trigger it on tags only.

### CODE REVIEW

File: conda/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+```
+
+When installing, to prioritize the CPG package, list the `cpg` channel before `bioconda`:
+
+```
+conda create --name hail -c cpg -c bioconda -c conda-forge hail
+conda activate hail
+```
+
+You can also install Hail into an existing enviornment; however note that Hail requires Python of versions 3.6 or 3.7, so Conda might downgrade Python in that environment, which may affect other installed packages.
```
Comment (lgruen) [inline conda/README.md:27]:
```suggestion
You can also install Hail into an existing environment; however note that Hail requires Python of versions 3.6 or 3.7, so conda might downgrade Python in that environment, which may affect other installed packages.
```

File: conda/hail/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
+{% set name = 'hail' %}
+{% set version = '0.2.61' %}
```
Comment (lgruen) [inline conda/hail/meta.yaml:2]: How does versioning work here? Shouldn't the package include the commit hash as the version or so, given that we would only increment the semantic version (0.2.61) whenever that happens upstream?

Comment (vladsaveliev) [reply conda/hail/meta.yaml:2]: I currently set it up to override the existing package in the channel. But good point, it should be better to append the commit hash!

File: conda/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Conda package
+
+This folder contains a Conda recipe to build the `hail` package for the [`cpg` Anaconda channel](https://anaconda.org/cpg/hail).
+
+Note that there is also a `hail` package in the [`bioconda` channel](https://github.com/bioconda/bioconda-recipes/tree/master/recipes/hail), which is synced with the [official PyPI release](https://pypi.org/project/hail). The idea of having a separate package in the `cpg` channel is to be able to build it from the most recent development codebase, that would reflect changes that were not yet released, or not to be propagated to the upstream repository at all.
```
Comment (pdiakumis) [inline conda/README.md:5]:
```suggestion
Note that there is also a `hail` package in the
[`bioconda` channel](https://github.com/bioconda/bioconda-recipes/tree/master/recipes/hail)
which is synced with the
[official PyPI release](https://pypi.org/project/hail).
Having a separate conda package in the `cpg` channel built from the
most recent development codebase allows us to test changes
that have not been released yet, or even use features that will not be
propagated to the upstream repository at all.
```

**Review by pdiakumis — APPROVED** [review]
This is awesome, thanks Vlad!

File: conda/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# Conda package
+
+This folder contains a Conda recipe to build the `hail` package for the [`cpg` Anaconda channel](https://anaconda.org/cpg/hail).
```
[the remark this answers is not in the document]
Comment (vladsaveliev) [reply conda/README.md]: Agree!

File: conda/README.md (hunk 3 above)
Comment (vladsaveliev) [reply conda/README.md]: Thanks!!

**Review by lgruen — APPROVED** [review]
This is great, thank you!

**Pull request merged**